src/string.cr

```diff
@@ -4244,8 +4244,11 @@ class String
   # Raises an `ArgumentError` if `self` has null bytes. Returns `self` otherwise.
   #
   # This method should sometimes be called before passing a `String` to a C function.
-  def check_no_null_byte
-    raise ArgumentError.new("String contains null byte") if byte_index(0)
+  def check_no_null_byte(name = nil)
+    if byte_index(0)
+      name = "`#{name}` "
+      raise ArgumentError.new("String #{name}contains null byte")
```

Comment:
Why is this on two lines??

Also, I'd rather the entire exception message was passed in, i.e. `def check_no_null_byte(message = "String contains null byte")`.

Reply:
The only thing you save by doing it the other way is a bit of typing, and the benefit of having fully customized error messages. For "Key contains null byte" vs "String \`key\` contains null byte" - the former is way better. Please, let's keep it simple and flexible.

Reply:
Two lines because I forgot a `if name` in the first line. To avoid empty argument to be printed.

Maybe full message is better. But actually, I don't think there is much benefit from it. Yes, it's more flexible, but I don't see any use for that. `check_no_null_byte` is only used to validate method arguments and for such it makes sense to declare the name of the argument if there are more than one. But apart from that, I don't see any need for customization.